### ISSUE TITLE
Discussion on the price slippage logic.  

### ISSUE THREAD

**Issue by xz-cn** [AcalaNetwork/acala-dapp#136]
I noticed if I set the price slippage to be 99%, I will get 0.001174 dot for 100aUSD, about 1% of the dot I get when set the slippage to be 1%.

![image](https://user-images.githubusercontent.com/46279370/88502996-dfb64d80-d002-11ea-92b9-a5738bbfa534.png)
![image](https://user-images.githubusercontent.com/46279370/88503003-e3e26b00-d002-11ea-9641-45c2640f34c2.png)

I understand the concept of price slippage is there to make sure a transaction can succeed when the price fluctuate. [Slippage can occur at any time but is most prevalent during periods of higher volatility when market orders are used. It can also occur when a large order is executed but there isn't enough volume at the chosen price to maintain the current bid/ask spread.](https://www.investopedia.com/terms/s/slippage.asp) 

But when the price is not fluctuating or there is enough volume in the pool, the dex should make sure the user have a transaction at the **highest** possible price, not the **lowest** one. So in my case, although I have singled that I accept a price slippage of 99%, the dex should still try to get me a best price (~0.11 dot) instead of the lowest possible price (~0.0017 dot).   

I think some discussion is needed for us to understand the price slippage logic in Acala and make sure it is implemented as intended. 

5GMvp5okVesv3JHxNA21SeFbXtMMtwy6EmPvtN5HpHLUuRCy

**Comment by qwer951123** [comment]
Sure, the slippage sets the minimum amount you accept but the Dex will try the best price when swap. when the slippage is 99%, that means 0.00117 is the lower bound if the result is small than 0.00117, this TX will failure.

**Comment by xz-cn** [comment]
> Sure, the slippage sets the minimum amount you accept but the Dex will try the best price to swap. when the slippage is 99%, that means 0.00117 is the lower bound if the result is small than 0.00117, this TX will failure.

Yes, 0.00117 dot is the lowest bound the user accepts. But I think the dex should still provide 0.117 dot if it is possible (in current case, it is actually possible).
